Continuous training

Andrea-Papaleo opened the pull request:
closes issue #511

Andrea-Papaleo committed:
[mod] store updates
 - changes how filtered categories etc. are handled in store
 - renames store slice types
 - conforms applicationSettingsSlice
 - creates custom checkbox and countChip components

Andrea-Papaleo committed:
[mod] reorganized store directory

Andrea-Papaleo committed:
[mod] groups ToolDrawer and ToolOptionDrawer
- updates and uses dimensions.ts

Andrea-Papaleo committed:
[mod] disables ripple on all buttons

Andrea-Papaleo committed:
[mod] creates ImageToolDrawer
 - creates filter tool

Andrea-Papaleo committed:
[mod] modifies filter list style
- adds collapse around chips for easing

Andrea-Papaleo committed:
[mod] updates classifier pipeline for continuous training
 - adds  and applies"unassigned" partition
 - updates partitions on labelling

Andrea-Papaleo committed:
[mod] update filter functionality
 - chips show visible properties
 - removes console logs
 - removes image tool draw in productiuon

gnodar01 committed:
[clean] Remove floating log

gnodar01 reviewed (changes requested):
Seems our diagram didn't cover an edge case:

- load plp1
- categorize *n* number of images as unknown
- enable shuffle on split
- train for some number of epochs
- predict
- don't recategorize anything
- hit clear predictions

now we have *n* number of uncategorized, unassigned images

That's exactly how we had it in our original diagram, but now we're in a situation where the user can't hit predict, because the inference set is empty. Similarly if additional images are uploaded, and given the unknown category, we cannot predict on them. The user's not completely stuck, since they can reassign the *n* images to the inference partition by training at least 1 epoch, but they should be able to hit predict without doing that. Additionally, in the case where the user first uploads *n* images, gives them the unknown category, and then **loads** a model, they can't hit predict since the inference set is empty, and they may not even be able to train in the case of a graph model.

I think we should just make it so that Uncategorized implies Inference, always. So the diagram would become:

![Andrea Papaleo's Zoom Meeting 2023-10-18 12_03 PM Page-1- 1702423238191](https://github.com/piximi/piximi/assets/8888732/0fc6b101-94d5-4f0a-8844-fe35d743f369)

gnodar01 reviewed (changes requested):
Tests seem to be failing

Andrea-Papaleo committed:
[mod] updated tests to reflect changes to entity adapters

Andrea-Papaleo committed:
[fix] fixes architecture settings not updating
 - CustomNumberTextField dispatchCallback

Andrea-Papaleo committed:
Merge branch 'continuous-training' of github.com:piximi/piximi into continuous-training

Andrea-Papaleo committed:
[mod] fixes issue #510
 - changes logic in ExampleProjectCard
 - project zip updated externally in S3

gnodar01 reviewed (changes requested):
BUG: Clicking "Hide labeled image" correctly dispatches updateImages with `visible: false` on labeled images, but the UI still displays the labeled images.

BUG: Edge case described earlier unresolved

Andrea-Papaleo committed:
[fix,mod] fixed hide/show labeled images
 - issue introduced in filtering logic in image selector
 - updates partition filter list when clicked, instead of image visibility

Andrea-Papaleo committed:
[mod,fix] fixes pedicting on newly uploaded images or double predict

Andrea-Papaleo committed:
[mod] modified update category button

Andrea-Papaleo committed:
[mod] renames ShowPredictionItems to PredictionListItems

Andrea-Papaleo committed:
[add,mod] adds accept all predictions button
 - renames ShowPredictions component
 - adds delayed list item button

gnodar01 reviewed (commented):
The following story causes an unexpected outcome
- Open mnist
- Train
- Predict
- Hide labeled image
- Accept all

You are then met with a blank screen.

Toggling Hide All / Show All unexpectedly does nothing.

This is not technically a dead state because what's actually going on is that only the "Inference partition" is being filtered for. So the user can hit "filter", press "x" on Inference under Image Partitions, causing the "training" and "validation" partitions to reappear.

I don't think this is a bug, but I think a more sensible default, after "accept all" is dispatched, is to automatically re-enable the other partitions.

Andrea-Papaleo committed:
[mod] resets partition filters when accepting predictions
